### PR TITLE
SMB share automation scripts

### PR DESCRIPTION
**Pull request by Wiezzel**
#### Phase 1: Check if the share is created and permissions granted
1. Create user `golem-docker` with password `golem-docker` (`New-LocalUser` in powershell - requires admin).
2. Create a test directory: `mkdir C:\Users\Public\test`
3. Create a sub-directory:  `mkdir C:\Users\Public\test\test1`
4. Create a text file with some data inside: `"test data" | Out-File -Encoding UTF8 C:\Users\Public\test\test1\data.txt`
4. Enable PowerShell scripting: `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned` (requires admin).
5. Go to golem repository and run the script: `python golem/docker/smbshare.py golem-docker C:\Users\Public\test`
6. You should see the following prompt. Click 'Yes'.
obraz
7. Script should finish without displaying any error messages.
8. Check if the share has been created. Running `Get-SmbShare` should print the following:
```
Name                             ScopeName Path                 Description
----                             --------- ----                 -----------
18E0021BB135EE9903D6BB747CD93B31 *         C:\Users\Public\test
```
9. Check (clicking 'properties' in Windows explorer) if user `golem-docker` has full permissions to `C:\Users\Public\test\test1` and `C:\Users\Public\test\test1\data.txt`.

#### Phase 2: Check if the created share is mountable in a container
1. Make sure you have Docker for Windows installed.
2. Check if docker-machine version is 0.15.0: `docker-machine --version`
2. Make sure you have and external virtual network switch ([howto](https://docs.docker.com/machine/drivers/hyper-v/#2-set-up-a-new-external-network-switch-optional)).
3. Create a VM: `docker-machine create --driver=hyperv --hyperv-boot2docker-url=https://github.com/golemfactory/boot2docker/releases/download/v18.06.0-ce%2Bdvm-v0.35/boot2docker.iso --hyperv-memory=2048 --hyperv-disable-dynamic-memory --hyperv-cpu-count=2 test`
4. Set environment variables to communicate with the machine: `docker-machine env test | Invoke-Expression`
5. Start a docker container in interactive mode with the directory mounted: `docker run -it --rm --mount volume-driver=cifs,src=$env:COMPUTERNAME/18E0021BB135EE9903D6BB747CD93B31/test1,target=/mnt/share,volume-opt=username=golem-docker,volume-opt=password=golem-docker ubuntu /bin/bash`
6. Once connected to the container manipulate some files in the mounted directory:
```
cd /mnt/share
cat data.txt
 test data
rm data.txt
mkdir foo
echo "test result" > foo/result.txt
```
7. End interactive session (<kbd>Ctrl</kbd>+<kbd>D</kbd>) and check if filesystem changes are persistent.

### CODE REVIEW

**Comment by Wiezzel** [comment]
@mfranciszkiewicz, @maaktweluit Plz test & review.

File: golem/docker/smbshare.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            '-File', SCRIPT_PATH,
+            '-UserName', user_name,
+            '-SharedDirPath', shared_dir_path],
+            timeout=SCRIPT_TIMEOUT,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT
+        )
+    except (CalledProcessError, TimeoutExpired) as exc:
+        raise RuntimeError(exc.stdout.decode('utf8'))
```
Comment (maaktweluit) [inline golem/docker/smbshare.py:30]: is there a difference between `stderr` and `stdout` on windows?

if so i think `stderr` is better to log as error, maybe `stdout` as debug

**Review by maaktweluit — CHANGES_REQUESTED** [review]
LGTM!

Should this `.ps1` file be included in the binary? 
then it has to be added in a pyinstaller hook

File: golem/docker/smbshare.py (hunk 1 above)
Comment (Wiezzel) [reply golem/docker/smbshare.py:30]: `stderr` is redirected to `stdout` See line 25.

**Comment by Wiezzel** [comment]
@maak Yep, the powershell script should be included.

**Review by maaktweluit — APPROVED** [review]

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**